packages/cli/src/index.js

```diff
@@ -31,69 +32,117 @@ import * as typeCheckCommand from './commands/type-check'
 import * as upgradeCommand from './commands/upgrade'
 import { getPaths } from './lib'
 
-/**
- * The current working directory can be set via:
- * 1. A `--cwd` option
- * 2. The `RWJS_CWD` env-var
- * 3. Found by traversing directories upwards for the first `redwood.toml`
- *
- * This middleware parses, validates, and sets current working directory
- * in the order above.
- */
-const getCwdMiddleware = (argv) => {
-  let configPath
-
-  try {
-    let cwd
-    if (argv.cwd) {
-      cwd = argv.cwd
-      // We delete the argument because it's not actually referenced in CLI,
-      // we use the `RWJS_CWD` env-var,
-      // and it conflicts with "forwarding" commands such as test and prisma.
-      delete argv.cwd
-    } else if (process.env.RWJS_CWD) {
-      cwd = process.env.RWJS_CWD
-    } else {
-      cwd = path.dirname(getConfigPath())
+// # Setting the CWD
+//
+// The current working directory can be set via:
+//
+// 1. The `--cwd` option
+// 2. The `RWJS_CWD` env-var
+// 3. By traversing directories upwards for the first `redwood.toml`
+//
+// ## Examples
+//
+// ```
+// yarn rw info --cwd /path/to/project
+// RWJS_CWD=/path/to/project yarn rw info
+//
+// # In this case, `--cwd` wins out over `RWJS_CWD`
+// RWJS_CWD=/path/to/project yarn rw info --cwd /path/to/other/project
+//
+// # Here we traverses upwards for a redwood.toml.
+// cd api
+// yarn rw info
+// ```
+
+let { cwd, testEnv } = Parser(hideBin(process.argv))
+cwd ??= process.env.RWJS_CWD
+
+try {
+  if (cwd) {
+    // `cwd` was set by the `--cwd` option or the `RWJS_CWD` env var. In this case,
+    // we don't want to find up for a `redwood.toml` file. The `redwood.toml` should just be in that directory.
+
+    const redwoodTOMLPath = path.join(cwd, 'redwood.toml')
```

Comment:
Just noticed I have one var called `redwoodTOMLPath` and another called `redwoodTomlPath`... will fix that